libgeopmd/src/LevelZero.cpp

```diff
@@ -481,53 +481,80 @@ namespace geopm
-            //       labeled as being either "compute" or "memory". So all the counters 
-            //       for now have been assigned L0_domain_type = M_DOMAIN_ALL 
-            m_devices.at(device_idx).subdevice.ras_domain.resize(geopm::LevelZero::M_DOMAIN_SIZE);
+            // Allocate size for a 2D vector to store all the RAS domain handles for errorsets:
+            //       m_num_subdevice = number of subdevices on specific GPU device
+            //       M_NUM_ERROR_TYPE = number of RAS error types
 
-            // Iterate over error set handles
+            m_devices.at(device_idx).subdevice.ras_domain.resize(num_subdevices * M_NUM_ERROR_TYPE);
+
+            // Iterate over errorset handles
             for (auto handle : error_set) {
-                // Get properties corresponding to each error set handle
+                // Get properties corresponding to each RAS errorset handle
                 zes_ras_properties_t property;
                 ze_result = zesRasGetProperties(handle, &property);
                 check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
                                 "LevelZero::" + std::string(__func__) +
                                 ": Sysman failed to get RAS properties",
                                 __LINE__);
 
+		// Check if the RAS errorset handle maps to a subdevice
                 if (property.onSubdevice == 0) {
 #ifdef GEOPM_DEBUG
                     std::cerr << "Warning: <geopm> LevelZero: A device level "
                               << "RAS domain was found but is not currently supported.\n";
 #endif
                 }
-                else {
-                    m_devices.at(device_idx).
-                        subdevice.ras_domain.at(geopm::LevelZero::M_DOMAIN_ALL).push_back(handle);
+		// Check if the RAS errorset handle maps to a known subdevice
+		else if (property.subdeviceId >= num_subdevice) {
+#ifdef GEOPM_DEBUG
+                    std::cerr << "Warning: <geopm> LevelZero: A RAS domain handle "
+                              << "was found to map to an unaccounted subdevice #"
+                              << property.subdeviceId << "\n";
+#endif
+
+                }
+		// Cache the RAS errorset handle into index - [subdevice Id][error type]
+                else if (property.type == ZES_RAS_ERROR_TYPE_CORRECTABLE ||
+                         property.type == ZES_RAS_ERROR_TYPE_UNCORRECTABLE) {
+                   int ras_idx = property.subdeviceId * M_NUM_ERROR_TYPE +
+                                 (property.type == ZES_RAS_ERROR_TYPE_CORRECTABLE ?
+                                  M_ERROR_TYPE_CORRECTABLE : M_ERROR_TYPE_UNCORRECTABLE);
+                   m_devices.at(device_idx).subdevice.ras_domain.at(ras_idx) = handle;
+                   ++ras_handle_count;
+#ifdef GEOPM_DEBUG
+		   std::cerr << "Level-0 RAS: Inserted handle for subdevice# "<<property.subdeviceId<<", error type# "<<property.type<<", device# "<<device_idx<<"\n";
```

Comment:
```suggestion
		   std::cerr << "Note: <geopm> Level-0 RAS: Inserted handle for subdevice# "<<property.subdeviceId<<", error type# "<<property.type<<", device# "<<device_idx<<"\n";
```
Should make it clear this debug note comes from GEOPM. Or if it was only there for initial development debug, maybe delete it.

Reply:
Good catch. This was debugging code intended only for initial development. Deleting line.